keanu-python/keanu/tensor.py

```diff
@@ -29,24 +29,16 @@ def __init__(self, t: tensor_arg_types) -> None:
 
     @staticmethod
     def __get_tensor_from_ndarray(ndarray: numpy_types) -> Any:
-        normalized_ndarray = Tensor.__ensure_rank_is_atleast_two(ndarray)
 
-        ctor = Tensor.__infer_tensor_ctor_from_ndarray(normalized_ndarray)
-        values = k.to_java_array(normalized_ndarray.flatten().tolist())
-        shape = k.to_java_long_array(normalized_ndarray.shape)
+        ctor = Tensor.__infer_tensor_ctor_from_ndarray(ndarray)
+        values = k.to_java_array(ndarray.flatten().tolist())
+        shape = k.to_java_long_array(ndarray.shape)
 
         return ctor(values, shape)
 
-    @staticmethod
-    def __ensure_rank_is_atleast_two(ndarray: numpy_types) -> numpy_types:
-        if len(ndarray.shape) == 1:
-            return ndarray[..., None]
-        else:
-            return ndarray
-
     @staticmethod
     def __infer_tensor_ctor_from_ndarray(ndarray: numpy_types) -> Any:
-        if len(ndarray) == 0:
+        if isinstance(ndarray, runtime_numpy_types) and len(ndarray) == 0:
```

Comment:
Why is this check (`isinstance(ndarray, runtime_numpy_types)`) needed still?

Reply:
It came over in a merge conflict. I've remove it.